src/parquet/arrow/writer.cc

```diff
@@ -548,6 +549,37 @@ Status FileWriter::Impl::TypedWriteBatch<ByteArrayType, ::arrow::BinaryType>(
   return Status::OK();
 }
 
+template <>
+Status FileWriter::Impl::TypedWriteBatch<FLBAType, ::arrow::FixedSizeBinaryType>(
+    ColumnWriter* column_writer, const std::shared_ptr<Array>& array, int64_t num_levels,
+    const int16_t* def_levels, const int16_t* rep_levels) {
+  RETURN_NOT_OK(data_buffer_.Resize(array->length() * sizeof(FLBA)));
```

Comment:
Pass `false` as the second parameter to Resize to avoid realloc https://github.com/apache/arrow/blob/master/cpp/src/arrow/buffer.h#L134